### DISCUSSION TITLE
Multiple packages in current directory == `cannot find package`

### DISCUSSION THREAD

**Discussion by jpluscplusm** [cue-lang/cue#1237] [Q&A] [answered]
Hey all 👋 😄 

Rather than kick off a GH Issue, I thought I'd ask here first ... perhaps it'll help someone in the future, too :-)

`cue` (v0.4.0) knows that it's found multiple packages in files in the current directory, and that that's a problem:

```
$ tail package-*
==> package-bar.cue <==
package bar

count: string
==> package-foo.cue <==
package foo

count: int
$ cue def
found packages "bar" (package-bar.cue) and foo (package-foo.cue) in "/home/user/tmp/cue-foo"
$ echo $?
1
```

... but asking it to do anything with a specific package fails:

```
$ cue def bar
cannot find package "bar"
$ cue def foo
cannot find package "foo"
$ cue def does_not_exist
cannot find package "does_not_exist"
```

What gives? What am I doing wrong? This behaviour persists regardless of the presence and contents of `cue.mod/module.cue`. 

TVM!

**Comment by verdverm** [comment]
See the `-p` flag. You can also specify the files to use as args

**Reply by jpluscplusm** [reply]
Heya 👋 Thanks for the reply :-) 

> See the -p flag. You can also specify the files to use as args

My reading of `-p` is that it applies to non-CUE files, so I don't /think/ that's what I'm looking for, here. It doesn't /seem/ to help, no matter how I introduce it:

```
$ cue def
found packages "bar" (package-bar.cue) and foo (package-foo.cue) in "/home/user/tmp/cue-foo"
$ cue def bar
cannot find package "bar"
$ cue def -p bar
found packages "bar" (package-bar.cue) and foo (package-foo.cue) in "/home/user/tmp/cue-foo"
$ cue def -p bar bar
cannot find package "bar"
$ cue def bar -p bar 
cannot find package "bar"
```

> You can also specify the files to use as args

So right now I'm using CUE in a package-less world, doing exactly what you mention: explicitly referring to all relevant files on all `cue` CLI invocations.

As I start to move towards the (hopefully!) sunnier uplands of "using packages", that means I *should* be able to split out CUE into multiple files which the loader *should* "just" find for me. If I stay with explicit references to each and every file I need, I can't really step off the iceberg into the lifeboat of "let CUE do the undifferentiated heavy lifting for me" ... :-)

If it's an unsupported mode to have files making up mutliple packges in the same directory, then that's ... worth knowing. I don't **believe** that's the case, as per https://cuelang.org/docs/concepts/packages/#overview: 

> There can be more than one package per directory

... but I can't figure out how to drive the CLI at my very early stage of moving-to-packages, so I thought I should validate some of my assumptions first!

**Reply by verdverm** [reply]
ah yes, I forget that the package doesn't work for selection. Intuitively it seems like it should and would be helpful. There is still inconsistency with the flags and commands.

You can use `hof mod` for some interim dependency management (https://github.com/hofstadter-io/hof). Private module support is close to being merged, just need to finish up some docs. (see the open PRs) 

There are also some related issues & discussions like #411 and something I can't find about using `if` to selectively import with injection / tags.

**Comment by antong** [comment, marked as answer]
Try `cue def .:foo` and `cue def .:bar`.  `cue help inputs`, explains a bit, but is missing some info.

**Reply by jbcpollak** [reply]
since upgrading from cue 0.2 to 0.4.1 I have the same problem, but with only one file in the directory:

```
❯ ls
arms.cue
❯ cue def arms
cannot find package "arms"
❯ cue def .:arms
package arms

#Example: {
        model: string
}
```

**Reply by mpvl** [reply]
`arms` is not a correct way to refer to the package. One can use the fully qualified `cue def .:arms`, `cue def .` (if there is only one package), or `cue def` (`.` is the default argument).
